sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -288,7 +288,14 @@ abstract class StreamExecution(
       // `postEvent` does not throw non fatal exception.
       val startTimestamp = triggerClock.getTimeMillis()
       postEvent(
-        new QueryStartedEvent(id, runId, name, progressReporter.formatTimestamp(startTimestamp)))
+        new QueryStartedEvent(
+          id,
+          runId,
+          name,
+          progressReporter.formatTimestamp(startTimestamp),
+          sparkSession.sparkContext.getJobTags()
```

Comment:
How many elements are we anticipating to have here? The size of event should be considerably small enough.

Reply:
Thanks for explanation.